ConfigSpace/conditions.py

```diff
@@ -207,63 +214,64 @@ def __init__(self, child, parent, value):
                              (child.name, value, parent.name))
         self.value = value
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return "%s | %s != %s" % (self.child.name, self.parent.name,
                                   repr(self.value))
 
-    def _evaluate(self, value):
-        return value != self.value
-        
+    def _evaluate(self, value: Any) -> bool:
+        if not self.parent.is_legal(value):
+            return False
+
+        return self.parent.get_order(value) != self.parent.get_order(self.value)
+
+
 class LessThanCondition(AbstractCondition):
-    def __init__(self, child, parent, value):
-        if not isinstance(parent, (NumericalHyperparameter,
-                                   OrdinalHyperparameter)):
-            raise ValueError("Parent hyperparameter in a < condition must "
-                             "be a subclass of NumericalHyperparameter or "
-                             "OrdinalHyperparameter, but is %s" % type(parent))
+    def __init__(self, child: Hyperparameter, parent: Hyperparameter, value: Any) -> None:
```

Comment:
I think you can assume for each argument `value` that it is of type `Union[str, float, int]`